server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -0,0 +1,257 @@
+        if (healthNode == null) {
+            this.isHealthNode.set(false);
+            this.cancelJob();
+            return;
+        }
+        final boolean isCurrentlyHealthNode = healthNode.getId().equals(this.clusterService.localNode().getId());
+        final boolean prevIsHealthNode = this.getIsHealthNode();
+        if (prevIsHealthNode != isCurrentlyHealthNode) {
+            this.isHealthNode.set(isCurrentlyHealthNode);
+            if (this.getIsHealthNode()) {
+                // we weren't the health node, and now we are
+                maybeScheduleJob();
+            } else {
+                // we were the health node, and now we aren't
+                cancelJob();
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        SchedulerEngine engine = scheduler.get();
+        if (engine != null) {
+            engine.stop();
+        }
+    }
+
+    @Override
+    public void triggered(SchedulerEngine.Event event) {
+        if (event.getJobName().equals(HEALTH_PERIODIC_LOGGER_JOB_NAME)) {
+            if (this.getIsHealthNode()) {
+                this.callGetHealth();
+            }
+        }
+    }
+
+    protected Clock getClock() {
+        return Clock.systemUTC();
+    }
+
+    protected void callGetHealth() {
+        this.healthService.getHealth(this.client, null, false, 0, this.resultsListener);
+    }
+
+    protected boolean getIsHealthNode() {
+        return isHealthNode.get();
+    }
+
+    protected SchedulerEngine getScheduler() {
+        return this.scheduler.get();
+    }
+
+    /**
+     * Handle the result of the Health Service getHealth call
+     */
+    protected final ActionListener<List<HealthIndicatorResult>> resultsListener = new ActionListener<List<HealthIndicatorResult>>() {
+        @Override
+        public void onResponse(List<HealthIndicatorResult> healthIndicatorResults) {
+            HealthPeriodicLoggerResult result = new HealthPeriodicLoggerResult(healthIndicatorResults);
+            Map<String, Object> resultsMap = result.toMap();
+            // if we have a valid response, log in JSON format
+            if (resultsMap.size() > 0) {
+                ESLogMessage msg = new ESLogMessage().withFields(result.toMap());
+                logger.info(msg);
+            }
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            logger.warn("Health Service logging error:{}", e.toString());
+        }
+    };
+
+    /**
+     * Create the SchedulerEngine.Job if this node is the health node
+     */
+    private void maybeScheduleJob() {
+        if (this.getIsHealthNode() == false) {
+            return;
+        }
+
+        // don't schedule the job if the node is shutting down
+        if (isClusterServiceStoppedOrClosed()) {
+            logger.trace(
+                "Skipping scheduling a HealthService job due to the cluster lifecycle state being: [{}] ",
```

Comment:
Could we name `HealthService job` a bit more specifically? For example, `health periodic logger job` or something similar.